core-features/napi-preview-feature/utils.ts

```diff
@@ -0,0 +1,115 @@
+import execa from 'execa'
+import fs from 'fs'
+
+const defaultExecaOptions = {
+  preferLocal: true,
+  stdio: 'inherit',
+  cwd: __dirname,
+} as const
+function buildSchema(previewFeatures?: string[]) {
+  const datasource = `
+datasource db {
+  provider = "sqlite"
+  url      = "file:./dev.db"
+}`
+  const previewFeaturesStr = previewFeatures
+    ? `previewFeatures = ${previewFeatures.map((p) => `["${p}"]`)}`
+    : ''
+
+  const generator = `
+generator client {
+  provider        = "prisma-client-js"
+  binaryTargets = ["native"]
+  ${previewFeaturesStr}
+}`
+
+  const models = `
+model User {
+  id    Int    @id
+  name  String
+  email String
+  posts Post[]
+}
+
+model Post {
+  id     Int @id
+  title  String
+  User   User   @relation(fields: [userId], references: [id])
+  userId Int
+}`
+  const schema = `${datasource}${generator}${models}`
+  const files = fs.readdirSync('./')
+  fs.writeFileSync('./prisma/schema.prisma', schema, {
+    encoding: 'utf-8',
+  })
+}
+
+async function generate(env?: Record<string, string>) {
+  await execa('yarn', ['prisma', 'generate'], {
+    ...defaultExecaOptions,
+    env,
+  })
+}
+
+async function clean() {
+  fs.rmdirSync('./node_modules/prisma', { recursive: true })
+  fs.rmdirSync('./node_modules/@prisma', { recursive: true })
+  fs.rmdirSync('./node_modules/.prisma', { recursive: true })
+  if (fs.existsSync('./prisma/schema.prisma')) {
+    fs.unlinkSync('./prisma/schema.prisma')
+  }
+  if (fs.existsSync('./data.json')) {
+    fs.unlinkSync('./data.json')
+  }
+}
+
+async function install(env?: Record<string, string>) {
+  await execa('yarn', ['install', '--force'], {
+    ...defaultExecaOptions,
+    env,
+  })
+}
+async function version(env?: Record<string, string>) {
+  const result = await execa('prisma', ['-v'], {
```

Comment:
I think this is run via `yarn` - then adding  `-s` to that command might help to get more stable output (and hence snapshots) (even if Yarn versions change in the future).

Reply:
Added